### PR TITLE
gossip a NodeDescriptor instead of a net.Addr

### PR DESCRIPTION
**Pull request by tbg**
also removed some duplicate code.

related to #543.

### CODE REVIEW

File: kv/dist_sender.go [hunk 1]
```diff
@@ -173,17 +173,6 @@ func (ds *DistSender) verifyPermissions(method string, header *proto.RequestHead
 		})
 }
 
-// nodeIDToAddr uses the gossip network to translate from node ID
```
Comment (spencerkimball) [inline kv/dist_sender.go]: Nice. Happy to see this moved to a central location.

File: storage/store.go [hunk 2]
```diff
@@ -79,6 +80,10 @@ var (
 	}
 )
 
+func init() {
+	gob.Register(&NodeDescriptor{})
```
Comment (spencerkimball) [inline storage/store.go]: Let's locate this statement in range.go, along with the others related to gossip.

**Comment by spencerkimball** [comment]
LGTM

File: storage/store.go (hunk 2 above)
Comment (tschottdorf) [inline storage/store.go]: Done.

**Pull request merged**